pkgs/development/libraries/glibc/ld-fallback.patch

```diff
@@ -0,0 +1,275 @@
+From 5956982e611afe46f2aa6a1e241262379fdbcbb6 Mon Sep 17 00:00:00 2001
+From: Jan Malakhovski <oxij@oxij.org>
+Date: Wed, 9 Aug 2023 08:53:06 +0000
+Subject: [PATCH] Implement support for LD_FALLBACK_PATH env variable and
+ related command line options
+
+LD_FALLBACK_PATH has the same semantics as LD_LIBRARY_PATH but unlike
+LD_LIBRARY_PATH, instead of having the highest priority, it has the second-lowest
+(it gets used just before the default system directories).
+
+Thus, LD_FALLBACK_PATH provides a way to specify fallback library paths.
+
+Consider the following use cases:
+
+- Running a closed-source program (like Steam and its games) on an unsupported
+  distribution: closed-source programs commonly override LD_LIBRARY_PATH to
+  override system libraries with their own bundled versions, but they
+  frequently do it the wrong way by overriding it while ignoring the current
+  value of LD_LIBRARY_PATH, thus failing to pick up any non-system libraries
+  they don't bundle but require specific versions of.
+
+  LD_FALLBACK_PATH, being the low-priority option, can't be used to achieve
+  the same effect as LD_LIBRARY_PATH from the perspective of the closed-source
+  program's vendor, thus making it unlikely they would ever touch
+  LD_FALLBACK_PATH. But it can be used by other tools to supply needed default
+  libraries to the closed-source program in question.
+
+- Similarly, sometimes you just want to change the underlying library for a
+  set of programs without changing the system default. For instance, if you
+  have different GPUs requiring different libGL's (though, specifically for
```

Comment:
In case of nixpkgs, wouldn't it be an option in that case to wrap each program requiring a different libGL (using `makeWrapper`) setting LD_LIBRARY_PATH accordingly?

Reply:
When a Python script in a devShell of a nix flake calls `git`, solved from PATH, the git might be installed from NixOS, and requires a glibc version different from the nix flake's nixpkgs.

Reply:
The just pushed version has a better solution for this via Nixpkgs `config`.